Documentation/contributing/testing/scalability.rst

```diff
@@ -0,0 +1,294 @@
+
+Metrics-based testing and alerting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes, you might want to scrape additional targets during test execution on top of the default ones.
+In this case, you can simply create a Pod or Service monitor `example monitor <EXAMPLE_MONITOR_>`_.
+Then you need to pass it as additional argument to ClusterLoader2:
+
+.. code-block:: bash
+
+    ./clusterloader \
+    --prometheus-additional-monitors-path=../../.github/actions/cl2-modules/egw/prom-extra-podmons
+    ...
+
+Now you can use the additional metrics in your test, by leveraging regular ``GenericPrometheusQuery`` measurement.
+For example, Egress Gateway ensures that various percentiles of masquerade latency observed by client are
+`below specific thresholds <EGW_MASQ_METRICS_>`_. This can be achieved by following measurement in ClusterLoader2:
+
+.. code-block:: yaml
+
+  - Identifier: MasqueradeDelay{{ .metricsSuffix }}
+    Method: GenericPrometheusQuery
+    Params:
+      action: {{ .action }}
+      metricName: Masquerade Delay {{ .metricsSuffix }}
+      metricVersion: v1
+      unit: s
+      enableViolations: true
+      queries:
+      - name: P95
+        query: quantile(0.95, egw_scale_test_masquerade_delay_seconds_total{k8s_instance="{{ .instance }}"})
+        threshold: {{ $MASQ_DELAY_THRESHOLD }}
+
+
+Running tests in CI
+~~~~~~~~~~~~~~~~~~~
+
+Once you are happy with the test and validated it locally, you can create a PR with the test.
+You can base your Github workflow on the existing tests, or add test scenario to already existing workflow.
+
+
+Accessing test results from PR or CI runs
+"""""""""""""""""""""""""""""""""""""""""
+
+You can run the specific scalability or performance test in your PR, some example commands are:
+
+.. code-block:: bash
+
+    /scale-100
+    /scale-clustermesh
+    /scale-egw
+    /fqdn-perf
+
+After test run, all of the results will be saved in GS bucket.
+In workflow run, you will see a link to the test results, for example in `test run <TEST_RUN_>`_:
+
+.. code-block:: bash
+
+    EXPORT_DIR: gs://cilium-scale-results/logs/scale-100-main/1745287079
+
+To see how to install gsutil check `Install gsutil <GSUTIL_INSTALL>`_ section.
+To see the results, you can run:
+
+.. code-block:: bash
+
+    gsutil ls -r gs://cilium-scale-results/logs/scale-100-main/1745287079
+
+You can also copy results to your local machine by running:
+
+.. code-block:: bash
+
+    gsutil -m cp -r gs://cilium-scale-results/logs/scale-100-main/1745287079 .
+
+
+.. _perfdashdocs:
+
+Visualizing results in Perfdash
+"""""""""""""""""""""""""""""""
+
+Perfdash leverages exported results from ClusterLoader2 and visualizes them.
+Currently, we do not host publicly available instance of Perfdash.
+To visualize the results, please check the `Scaffoling repository <PERFDASH_>`_.
+As an example, you can check how average CPU usage of the Cilium agent changed recently:
```

Comment:
Please remove “changed recently”, this is only valid as of your writing, not in months or years from now when readers take a look at the docs.